tests/test_tk_docs_preview.py

```diff
@@ -12,63 +12,58 @@
 import os
 import tempfile
 
-from tk_toolchain.cmd_line_tools import tk_docs_preview
-
-CURRENT_REPO_ROOT = os.path.dirname(os.path.dirname(__file__))
-REPOS_ROOT = os.path.dirname(CURRENT_REPO_ROOT)
+import pytest
 
-TK_CONFIG_ROOT = os.path.join(REPOS_ROOT, "tk-config-basic")
-TK_CORE_ROOT = os.path.join(REPOS_ROOT, "tk-core")
-TK_FRAMEWORK_ROOT = os.path.join(REPOS_ROOT, "tk-framework-shotgunutils")
-PYTHON_API_ROOT = os.path.join(REPOS_ROOT, "python-api")
+from tk_toolchain.cmd_line_tools import tk_docs_preview
 
 
 # Note: These tests are likely to introduce side effects because they monkey
 # patch toolkit. For now we're running them in-process because it makes
 # coverage easier to retrieve
 
 
-def test_without_any_parameters():
+def test_without_any_parameters(tk_framework_root):
     """
     Make sure we can generate documentation from inside a repository
     without any arguments.
     """
     cwd = os.getcwd()
     # Switch current folder to the framework so we can test both the core and
     # framework detection code path.
-    os.chdir(TK_FRAMEWORK_ROOT)
+    os.chdir(tk_framework_root)
     try:
         assert tk_docs_preview.main(["tk_docs_preview", "--build-only"]) == 0
     finally:
         os.chdir(cwd)
 
 
-# Disabling this test as the doc for the Python API is currently broken!
-def _test_with_python_api():
+@pytest.mark.xfail(reason="Documentation is currently broken for this tool.")
```

Comment:
Mark the test as expected to fail, so having it fail will not fail the test suite.